oauth2client/contrib/gce.py

```diff
@@ -51,13 +59,17 @@ class AppAssertionCredentials(AssertionCredentials):
     """
 
     @util.positional(2)
-    def __init__(self, scope, **kwargs):
+    def __init__(self, scope='', **kwargs):
         """Constructor for AppAssertionCredentials
 
         Args:
             scope: string or iterable of strings, scope(s) of the credentials
```

Comment:
I just realized the scope (heh) of this change is larger than I had been thinking, because this change doesn't eliminate the assignment of the instance-scope `scope` field from the instance constructor.

This means we're dug into a hole of some sort, right? And of some depth?

How about an implementation comment explaining why it is that users must not pass a non-empty value of the `scope` parameter but we're forced to assign it to an instance-scope attribute?

Is it because the attribute is a public attribute?

Apologies for the confusion. Also I'll be away from my screen for the next few hours.